extensions/firestore-sheets-sync/functions/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import * as functions from 'firebase-functions';
+import { GaxiosError } from 'gaxios';
+import { google, sheets_v4 } from 'googleapis';
+
+import config from './config';
+
+const sheets = google.sheets('v4');
+const authClient = google.auth.getClient({
+  // keyFilename: 'extensions-testing-cae073db7903.json',
+  scopes: ['https://www.googleapis.com/auth/spreadsheets'],
+});
+
+const fields = config.fields.split(',');
+
+export const googleSheetSync = functions.firestore
+  .document(`${config.collection}`)
+  .onWrite(async change => {
+    // Get the document that was written to Firestore, and structure it.
+    const data = change.after.data();
+
+    if (!data) return;
+
+    await createHeaderRow();
+
+    const rows: any[] = [];
+
+    console.log(fields);
+
+    for (const field of fields) {
+      const value = data[field];
+      if (value) rows.push(value);
+      else rows.push('');
+    }
+
+    try {
+      // Append the data to the Google Sheet.
+      await appendNewRow(rows);
+
+      return;
+    } catch (error) {
+      // Log an error if the data was not written to the Google Sheet.
+      functions.logger.error((error as GaxiosError).message, error);
+    }
+  });
+
+async function appendNewRow(data: any[]) {
+  return sheets.spreadsheets.values.append({
+    auth: await authClient,
+    spreadsheetId: config.spreadsheetId,
+    valueInputOption: 'RAW',
+    requestBody: { values: [data], range: 'A2:Z2' },
+    range: 'A2:Z2',
```

Comment:
```suggestion
    requestBody: { values: [data], range: '2:2' },
    range: '2:2',
```

Can we use the row notation?